app/controllers/road-sign-concepts/road-sign-concept.js

```diff
@@ -1,32 +1,71 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
-import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
 
 export default class RoadsignConceptsRoadsignConceptController extends Controller {
-  @service router;
-  @service intl;
+  @tracked isToShowSubSigns = false;
+  @tracked isToShowRelated = false;
+
+  @tracked category = null;
 
   @action
-  async removeRoadSignConcept(roadSignConcept, event) {
+  async editSubSignRelation(subSign, event) {
     event.preventDefault();
-
-    await roadSignConcept.destroyRecord();
-    this.router.transitionTo('road-sign-concepts');
+    let subSigns = await this.model.roadSignConcept.subSigns;
+    let roadSigns = await this.model.allSubSigns.roadSignConcepts;
+    if (event.target.checked) {
+      subSigns.pushObject(subSign);
+      this.model.roadSignConcept.save();
+      roadSigns.removeObject(subSign);
```

Comment:
Wait, this will update the actual database? I think this will remove the relationship between the road sign and its category, no? I think you only need to update your subSigns relationship here. I'm not sure If I understand what's happening.

Reply:
Yes, you're right, but maybe it has to update a list that is a copy from allSubSigns to not have duplications when select.